Library/MapleBacon/MapleBacon/Resizer.swift

```diff
@@ -6,176 +6,251 @@ import UIKit
 
 public typealias ResizerCompletion = (UIImage) -> Void
 
-public class Resizer {
-
-    private static let deviceScale = UIScreen.mainScreen().scale
+internal extension CGSize {
+    
+    func scaled(factor: CGFloat? = nil) -> CGSize {
+        let scale = factor ?? UIScreen.main().scale
+        return CGSize(width: self.width * scale, height: self.height * scale)
+    }
+    
+    func zeroBoundedRect() -> CGRect {
+        return CGRect(x: 0, y: 0, width: self.width, height: self.height)
+    }
+}
 
-    public class func resizeImage(image: UIImage, toSize size: CGSize, async: Bool = true, completion: ResizerCompletion) {
-        resizeImage(image, contentMode: .ScaleToFill, toSize: size, interpolationQuality: .Default,
-                async: async, completion: completion)
+internal extension CGRect {
+    
+    func scaled(factor: CGFloat? = nil) -> CGRect {
+        let scale = factor ?? UIScreen.main().scale
+        return CGRect(x: self.origin.x * scale, y: self.origin.y * scale, width: self.size.width * scale, height: self.size.height * scale)
     }
+    
+    func zeroBoundedRect() -> CGRect {
+        return CGRect(x: 0, y: 0, width: self.size.width, height: self.size.height)
+    }
+}
 
-    public class func resizeImage(image: UIImage, contentMode: UIViewContentMode, toSize size: CGSize,
-                                  interpolationQuality quality: CGInterpolationQuality,
-                                  async: Bool = true, completion: ResizerCompletion) {
+// MARK: - Resizer Class -
+public class Resizer {
+    
+    var image: UIImage
+    
+    /**
+     Internal entity to compute and hold resizing values
+    */
+    private struct ResizeCalculationEntity {
+        
+        var currentSize: CGSize  = CGSize()
+        var intendedSize: CGSize = CGSize()
+        
+        var horizontalRatio: CGFloat = 0
+        var verticalRatio: CGFloat = 0
+        var newSize: CGSize = CGSize()
+        var offset: CGSize = CGSize()
+        var offsetPoint: CGPoint = CGPoint()
```

Comment:
a little issue with the naming… `CGPoint` makes clear that it's a point, so this variable should be called just `offset` and the `CGSize` above maybe `offsetBy`